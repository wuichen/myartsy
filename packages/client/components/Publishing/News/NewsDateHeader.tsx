import { unica } from "Assets/Fonts"
import { pMedia } from "Components/Helpers"
import React from "react"
import styled from "styled-components"
import { getDate } from "../Constants"

interface Props {
  date: string
}

export const NewsDateHeader: React.SFC<Props> = props => {
  const { date } = props

  return <NewsText>{getDate(date, "news")}</NewsText>
}

export const NewsText = styled.div`
  ${unica("s25", "medium")};
  ${pMedia.sm`
    ${unica("s16", "medium")}
  `};
`
