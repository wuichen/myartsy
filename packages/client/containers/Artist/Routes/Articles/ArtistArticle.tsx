import { Col, Row } from "@artsy/palette"
import React, { SFC } from "react"
import { Media } from "Utils/Responsive"

import {
  Box,
  Flex,
  Image,
  Sans,
  Separator,
  Serif,
  Spacer,
} from "@artsy/palette"

interface ArticleItemProps {
  imageUrl: string
  date: string
  author: string
  title: string
  href: string
  lastChild: boolean
}

export const ArticleItem: SFC<ArticleItemProps> = props => {
  return (
    <>
      <Media at="xs">
        <SmallArticleItem {...props} />
      </Media>
      <Media greaterThan="xs">
        <LargeArticleItem {...props} />
      </Media>
    </>
  )
}

const LargeArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title, lastChild } = props

  return (
    <>
      <Row
        style={{ cursor: "pointer" }}
        onClick={() => {
          window.location.href = href
        }}
      >
        <Col sm={2}>
          <Date size="3" weight="medium">
            {date}
          </Date>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={12}>
              <Title size="5">
                <a href={href} className="noUnderline">
                  {title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {author}
              </Credit>
            </Box>
          </Flex>
        </Col>
        <Col sm={2}>
          <a href={href}>
            <Image
              style={{
                height: "100px",
                width: "auto",
                maxWidth: 140,
                maxHeight: 280,
                objectFit: "fill",
              }}
              src={imageUrl}
              mb={2}
            />
          </a>
        </Col>
      </Row>
      {/* FIXME: Weird block height issue... */}
      {!lastChild && <Separator />}
      <Spacer mt={3} />
    </>
  )
}

const SmallArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title, lastChild } = props

  return (
    <>
      <Flex
        justifyContent="space-between"
        style={{ cursor: "pointer" }}
        onClick={() => {
          window.location.href = href
        }}
      >
        <Box pr={2}>
          <Date size="1" weight="medium">
            {date}
          </Date>
          <Spacer mb={0.5} />
          <Title size="2">
            <a href={href} className="noUnderline">
              {title}
            </a>
          </Title>
          <Spacer mb={0.5} />
          <Credit size="1" color="black60">
            {author}
          </Credit>
        </Box>
        <Image width="70px" height="100%" src={imageUrl} />
      </Flex>
      {/* FIXME: Weird block height issue... */}
      <Spacer mt={2} />
      {!lastChild && <Separator />}
      <Spacer mt={3} />
    </>
  )
}

const Date = Sans
const Title = Serif
const Credit = Serif
