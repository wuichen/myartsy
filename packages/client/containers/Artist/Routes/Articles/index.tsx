import { Articles_artist } from "__generated__/Articles_artist.graphql"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import { ArtistArticlesRefetchContainer as Articles } from "./ArtistArticles"

export interface ArticlesRouteProps {
  artist: Articles_artist
}

export const ArticlesRoute: React.SFC<ArticlesRouteProps> = props => {
  return <Articles artist={props.artist} />
}

export const ArticlesRouteFragmentContainer = createFragmentContainer(
  ArticlesRoute,
  {
    artist: graphql`
      fragment Articles_artist on Artist {
        ...ArtistArticles_artist
      }
    `,
  }
)
