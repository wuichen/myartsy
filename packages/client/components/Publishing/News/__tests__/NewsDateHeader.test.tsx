import { mount } from "enzyme"
import "jest-styled-components"
import { DateTime } from "luxon"
import React from "react"
import renderer from "react-test-renderer"
import { NewsArticle } from "../../../../Components/Publishing/Fixtures/Articles"
import { NewsDateHeader } from "../NewsDateHeader"

describe("NewsDateHeader", () => {
  it("renders a news date header properly", () => {
    const component = renderer
      .create(<NewsDateHeader date={NewsArticle.published_at} />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  it("Renders 'today' if date is today", () => {
    const date = DateTime.local().toString()
    const wrapper = mount(<NewsDateHeader date={date} />)
    expect(wrapper.text()).toMatch("Today")
  })

  // FIXME: Figure out how to make dates pass CI
  xit("Renders date with no year if in current year", () => {
    const date = DateTime.local()
      .minus({ weeks: 1 })
      .toString()
    const wrapper = mount(<NewsDateHeader date={date} />)
    expect(wrapper.text()).toMatch(DateTime.fromISO(date).toFormat("MMM d"))
    expect(wrapper.text()).not.toMatch(DateTime.fromISO(date).toFormat("yyyy"))
  })

  xit("Renders date with year if not in current year", () => {
    const date = DateTime.local()
      .minus({ years: 1 })
      .toString()
    const wrapper = mount(<NewsDateHeader date={date} />)
    expect(wrapper.text()).toMatch(
      DateTime.fromISO(date).toFormat("MMM d, yyyy")
    )
  })
})
