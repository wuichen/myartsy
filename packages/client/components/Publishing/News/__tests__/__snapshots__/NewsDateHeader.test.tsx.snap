// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsDateHeader renders a news date header properly 1`] = `
.c0 {
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 25px;
  line-height: 1.1em;
}

@media (max-width:720px) {
  .c0 {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }
}

<div
  className="c0"
>
  Jul 19, 2018
</div>
`;
